File: client/src/app/modules/pao/services/journal.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Journal } from '../../pao/model/Journal';

@Injectable({
  providedIn: 'root'
})
export class JournalService extends AppRestService<Journal>{

  constructor(http: HttpClient) {
    super(http, "pao");
  }

  getJournals() {
    return this.getList(new HttpParams(), "journals");
  }

  getJournal(idJournal: number) {
    return this.getById("journal", idJournal);
  }

  addOrUpdateJournal(journal: Journal) {
    return this.addOrUpdate(new HttpParams(), "journal", journal, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/pao/service/JournalService.java
package com.jss.osiris.modules.pao.service;

import java.util.List;

import com.jss.osiris.libs.exception.OsirisClientMessageException;
import com.jss.osiris.libs.exception.OsirisException;
import com.jss.osiris.modules.pao.model.Journal;

public interface JournalService {
    public List<Journal> getJournals();

    public Journal getJournal(Integer id);

    public Journal addOrUpdateJournal(Journal journal) throws OsirisException, OsirisClientMessageException;
}
